cudamapper/src/index_gpu.cuh

```diff
@@ -519,11 +531,11 @@ namespace index_gpu {
 
             std::uint64_t total_basepairs = 0;
             std::vector<ArrayBlock> read_id_to_basepairs_section_h;
-
+            std::size_t global_read_id = 0;
             std::vector<std::size_t> fasta_indices;
             // find out how many basepairs each read has and determine its section in the big array with all basepairs
             for (std::size_t fasta_object_id = 0; fasta_object_id < fasta_objects.size(); ++fasta_object_id) {
-                auto global_read_id = current_chunk_start + fasta_object_id;
+                global_read_id = current_chunk_start + fasta_object_id;
```

Comment:
What the reason for moving it outside of the for loop if its value gets overwritten here?